src/base.js

```diff
@@ -364,14 +396,59 @@ export default class KintoClientBase {
       return raw ? { json: msg, headers: { get() {} } } : msg;
     }
     await this.fetchServerSettings();
-    const result = await this.http.request(this.remote + request.path, {
-      ...request,
-      body: stringify ? JSON.stringify(request.body) : request.body,
-    });
+    // FIXME: Because any fetch() option is accepted here, `request`
+    // could have lots of parameters that wouldn't be accepted as part
+    // of a batch request. This makes it easier to shoot yourself in
+    // the foot. We should probably only pass through the four
+    // parameters accepted in a batch request: method, body, path, and
+    // headers.
+    const result = await this.http.request(
+      this.remote + request.path,
+      {
+        ...request,
+        body: stringify ? JSON.stringify(request.body) : request.body,
+      },
+      {
+        retry: this._getRetry(options),
+      }
+    );
     return raw ? result : result.json;
   }
 
+  /**
+   * Fetch some pages from a paginated list, following the `next-page`
+   * header automatically until we have fetched the requested number
+   * of pages. Return a response with a `.next()` method that can be
+   * called to fetch more results.
+   *
+   * @private
+   * @param  {String}  path
+   *     The path to make the request to.
+   * @param  {Object}  params
+   *     The parameters to use when making the request.
+   * @param  {String}  [params.sort="-last_modified"]
+   *     The sorting order to use when fetching.
+   * @param  {Object}  [params.filters={}]
+   *     The filters to send in the request.
+   * @param  {Number}  [params.limit=undefined]
+   *     The limit to send in the request. Undefined means no limit.
+   * @param  {Number}  [params.pages=undefined]
+   *     The number of pages to fetch. Undefined means one page. Pass
+   *     Infinity to fetch everything.
+   * @param  {String}  [params.since=undefined]
+   *     The ETag from which to start fetching.
+   * @param  {Object}  [options={}]
+   *     Additional request-level parameters to use in all requests.
+   * @param  {Object}  [options.headers={}]
+   *     Headers to use during all requests.
+   * @param  {Number}  [options.retry=0]
+   *     Number of times to retry each request if the server responds
+   *     with Retry-After.
+   */
   async paginatedList(path, params, options = {}) {
+    // FIXME: this is called even in batch requests, which doesn't
+    // make any sense (since all batch requests get a "dummy"
+    // response; see execute() above).
```

Comment:
I'm not sure what the right fix is, but I don't want to preclude the possibility of a more involved refactor to turn e.g. Collections into BatchCollections which have different parameters and return different things.